lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -1156,7 +1157,7 @@ impl ModuleCodeGenerator<X64FunctionCode, X64ExecutionContext, CodegenError>
     unsafe fn from_cache(artifact: Artifact, _: Token) -> Result<ModuleInner, CacheError> {
         let (info, _, memory) = artifact.consume();
 
-        let cache_image: CacheImage = bincode::deserialize(memory.as_slice())
+        let cache_image: CacheImage = BorshDeserialize::deserialize(&mut memory.as_slice())
```

Comment:
Could we keep an option to use both deserialization protocols?

Reply:
Do you mean:
- first try deserialize with borsh
- if fail then try deserialize with bincode
- if still fail return error

Reply:
For me just `if global_use_borsch deserailize_borsh() else deserailize_serde()` would do.

Reply:
There is no way to pass `global_use_borsh` param into `from_cache`, unless it's a global option or a feature flag, but that would not make much sense if what we want is just enable borsh serializer. And to make current cache backward compatible, so i think it's better to:
- when serialize, always borsh serialize so new cache is borsh serialized
- when deserialize, try borsh, then bincode then error, so both old cache and new cache won't crash

Reply:
Actually my proposed approach is problematic, there can be deserialize with no error, but there's actual logical error of the deserialization result. But I still find that `global_use_borsh` option is not helpful because: we used hashed key which indicates cache version, there is no way to tell an arbitrary key is old version or new, and based on the version choose different deserialization method. So I just bumped the cache version to make new keys serialized with borsh and old cache won't be accessible. This way we don't need a global `global_use_borsh` option to control since every cache that can be accessed is borsh serialized and should be borsh deserialized